src/main/java/net/openhft/chronicle/queue/impl/single/StoreTailer.java

```diff
@@ -1112,9 +1118,12 @@ private boolean tryWindBack(final int cycle) {
         if (count <= 0)
             return false;
         final RollCycle rollCycle = queue.rollCycle();
-        moveToIndexInternal(rollCycle.toIndex(cycle, count - 1));
-        this.state = FOUND_IN_CYCLE;
-        return true;
+        if (moveToIndexInternal(rollCycle.toIndex(cycle, count - 1))) {
+            this.state = FOUND_IN_CYCLE;
+            return true;
+        } else {
+            return false;
+        }
```

Comment:
ONLY set the state to `FOUND_IN_CYCLE` after we've successfully loaded the cycle, we can't assume that because we were able to find the last index a moment ago, that the cycle is still there and loadable.